### PR TITLE
New editor open

### PR DESCRIPTION
**Pull request by facelessuser**
Also improve some window flickering and do better clean up on app closure. Ref #119.

### CODE REVIEW

**Comment by facelessuser** [comment]
In the editor edit dialog, we dynamically wrap the dialog content on resize.  This seems to work everywhere, but we seem to have a difficult time calculating the wrap length in Windows.  For all other systems, 20 seems to be enough.  This is gathered from the border margin of the containing panel and dialog, 5 on both the left and the right for each giving us a total of 20.  Windows seems to need 40 which is double.  I don't know if that is related to my Windows systems using 4K monitors set at a font scale of 2X or if this is just normal.  This needs to be investigated.  If all we need to do is `offset = 40 if windows else 20`, then that is fine, if not, I'd like to detect this or I'll just have to set 40 to everyone which is fine, but it will bug me a little...

**Comment by facelessuser** [comment]
On a side note, because I've adjusted some resize behaviors on dialogs in this pull and will probably just make the change here, I plan on making the main dialog  only change the dialog size on initial open opposed to all the time.  The dialog used to do this, but it was always a bit buggy.  I think I have code to do this, but I am having a difficult time with OSX fullscreen mode.  When this mode is entered, the proportions are different.  So if we hide the filter controls and change the minimum resize height, it is wrong when we exit fullscreen mode.  This is no big deal *except* wxPython can't seem to detect when we enter or exit fullscreen mode because the system is doing it opposed to toggling through their API.  Ugh.  I can't seem to disable fullscreen mode either.  I really don't want to check on every resize event to ensure our minimum size height is correct, but I may be running out of options.

Basically the issue is if you come out of fullscreen mode, you can resize the height slightly smaller than you should which bothers me...but maybe I should just overlook it

**Comment by facelessuser** [comment]
All previously known resize issues seemed fixed.  Unfortunately, linux has a maximized issue that behaves similar to the mac issue, only client size doesn't change, so it is harder to detect without on every resize calculating what it should be.  Hopefully I can come up with a better way than that.
